openpype/client/entities.py

```diff
@@ -32,17 +32,37 @@ def _prepare_fields(fields, required_fields=None):
     return output
 
 
-def _convert_id(in_id):
+def convert_id(in_id):
+    """Helper function for conversion of id from string to ObjectId.
+
+    Args:
+        in_id (Union[str, ObjectId, Any]): Entity id that should be converted
+            to right type for queries.
+
+    Returns:
+        Union[ObjectId, Any]: Converted ids to ObjectId or in type.
+    """
+
     if isinstance(in_id, six.string_types):
         return ObjectId(in_id)
     return in_id
```

Comment:
Shouldn't this also ensure the returned type is actually `ObjectId`.
If I'd pass an `int` or `float` to the function it would come out exactly the same as an `int` or `float`.

Also the `in_id` type hint thus seems to broad with `Any` in there. No?